### PR TITLE
Use same date within a single run

### PR DESCRIPTION
**Pull request by bakshiutkarsha**
should reolve #1178

### CODE REVIEW

**Comment by bakshiutkarsha** [comment]
> * This does not fix the problem I believe because the date will be still different.

For the first point:
`This does not fix the problem I believe because the date will be still different.
`
I believe it should work as I am creating an instance of date before the `for` loop begins to create zims of different flavours.

For the last point:
`We don't need an extra getZimCreationDate for something which should be use only one time and which is a one line function.`

I am creating a method for this because it is being used in multiple places in test cases. Should I use hardcoded value of dates in test cases or should I duplicate the code?

**Comment by kelson42** [comment]
> > * This does not fix the problem I believe because the date will be still different.
> 
> I believe it should work as I am creating an instance of date before the `for` loop begins to create zims of different flavours.

You call the method 3 times within the for loop, not before, see: https://github.com/openzim/mwoffliner/pull/1181/files#diff-4cc2b78cf5db017b0392c57c374d64dfR356. BTW, identation is wrong there.

> > * We don't need an extra getZimCreationDate for something which should be use only one time and which is a one line function
> 
> I am creating a method for this because it is being used in multiple places in test cases. Should I use hardcoded value of dates in test cases or should I duplicate the code?

Probably here it should be able to work (so not date in the filename) if not specified at all.

File: src/Dump.ts [hunk 1]
```diff
@@ -33,7 +34,6 @@ export class Dump {
     public novid: boolean;
     public nopdf: boolean;
     public nodet: boolean;
-    public contentDate: string;
```
Comment (kelson42) [inline src/Dump.ts:36]: `filenameDate`, not `contentDate` which is too vague/wrong.

File: src/mwoffliner.lib.ts [hunk 2]
```diff
@@ -325,6 +325,9 @@ async function execute(argv: any) {
     // await trimUnmirroredPages(downloader); // TODO: improve simplify graph to remove the need for a second trim
   }
 
+  const date = new Date();
+  const contentDate = `${date.getFullYear()}-${(`0${date.getMonth() + 1}`).slice(-2)}`;
```
Comment (kelson42) [inline src/mwoffliner.lib.ts:329]: Something like `const filenameDate = (new Date()).toISOString().slice(0, 7)` seems better to me. Only one line and simplier.

Comment (bakshiutkarsha) [reply src/mwoffliner.lib.ts:329]: `(new Date()).toISOString().slice(0, 7)`
 will give "2020-06", which is only year and month. Just to confirm we do want the date as well, right? If yes, I just need to change the slice param.

**Pull request merged**